### PR TITLE
Blog app: Add-API-Endpoints

### PR DESCRIPTION
**Pull request by Kumilachew-g**
**In this PR, We have done the following tasks.**
- Created an API endpoint to list all posts for a user.
- Created an API endpoint to list all comments for a user's post.
- Created an API endpoint to add a comment to a post. Remember that the owner of the comment is the user that makes it; it should not be set by passing a parameter.
- Added authentication to my API endpoints.
- My API endpoints could receive JSON and respond to JSON as well.

### CODE REVIEW

**Review by Davidosky007 — COMMENTED** [review]
Hi @Kumilachew-g @MrOmachi 👋, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that doesn't show contribution from your partner @MrOmachi as this is a paired programming activity you should work together 
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by Kumilachew-g** [comment]
Hi👋 @Davidosky007, We have done this based on your comment.

**Review by codecaiine — CHANGES_REQUESTED** [review]
Hi @Kumilachew-g @MrOmachi , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that I can't create a user and test all endpoints properly. Please make sure the user creation and make sure that all the endpoint work. :+1: 


Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by Kumilachew-g** [comment]
**It works for me just like this...**
1. 
image
3. 
image

File: app/controllers/api/comments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      @comments = Comment.where(user_id: @user.id)
+      respond_to do |format|
+        format.json { render json: @comments, status: :ok }
+      end
+    else
+      respond_to do |format|
+        format.json { render json: 'please sign in or add the token to create comment'.to_json, status: :ok }
+      end
+    end
+  end
```
Comment (EricMbouwe) [inline app/controllers/api/comments_controller.rb]: - [ ] You should have an API endpoint that return all comments for a post - **GET :** `/users/:user_id/posts/:id/comments`. Here you are missing a reference to the `post` where to fetch the list of comments.

Screenshot 2022-12-09 at 2 07 39 PM

**Review by EricMbouwe — CHANGES_REQUESTED** [review]
Hi @Kumilachew-g, 

Good job so far 🔥!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 🔥🎉
- All API endpoints receive JSON and respond in a JSON format as well. ✔
- API endpoint that returns all post in JSON format - **GET :** `/api/users/:user_id/posts` ✔
- API endpoint that adds' a comment to a post - **POST :** `/users/:user_id/posts/:id/comments` ✔
- API endpoint for authentication, both registration and login ✔
- PR ✔
- No linter errors ✔

# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/api/comments_controller.rb (hunk 1 above)
Comment (Kumilachew-g) [reply app/controllers/api/comments_controller.rb]: **I have done this based on review**

image

**Review by AnselemOdims — APPROVED** [review]
### Hi  @Kumilachew-g 🙋‍♀️, 
Good job in making the required changes 👏 

# STATUS APPROVED ✔
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Use the tag @AnselemOdims for a quicker reply


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**